test/foundry/BullaFrendLend/BullaFrendLend.t.sol

```diff
@@ -298,4 +298,321 @@ contract TestBullaFrendLend is Test {
         assertEq(offerCreditor, address(0), "Offer should be deleted after rejection");
     }
+
+    function testPartialLoanPayments() public {
+        vm.prank(creditor);
+        weth.approve(address(bullaFrendLend), 2 ether);
+        
+        vm.prank(creditor);
+        weth.approve(address(bullaClaim), 2 ether);
+        
+        vm.prank(debtor);
+        weth.approve(address(bullaClaim), 2 ether);
+        
+        vm.prank(debtor);
+        weth.approve(address(bullaFrendLend), 2 ether);
+        
+        LoanOffer memory offer = LoanOffer({
+            interestBPS: 500,
+            termLength: 30 days,
+            loanAmount: 1 ether,
+            creditor: creditor,
+            debtor: debtor,
+            description: "Partial Payments Test Loan",
+            token: address(weth)
+        });
+
+        uint256 initialCreditorWeth = weth.balanceOf(creditor);
+        uint256 initialDebtorWeth = weth.balanceOf(debtor);
+
+        vm.prank(creditor);
+        uint256 loanId = bullaFrendLend.offerLoan{value: FEE}(offer);
+
+        bullaClaim.permitCreateClaim({
+            user: debtor,
+            operator: address(bullaFrendLend),
+            approvalType: CreateClaimApprovalType.Approved,
+            approvalCount: 1,
+            isBindingAllowed: true,
+            signature: sigHelper.signCreateClaimPermit({
+                pk: debtorPK,
+                user: debtor,
+                operator: address(bullaFrendLend),
+                approvalType: CreateClaimApprovalType.Approved,
+                approvalCount: 1,
+                isBindingAllowed: true
+            })
+        });
+
+        vm.prank(debtor);
+        uint256 claimId = bullaFrendLend.acceptLoan(loanId, ClaimMetadata("", ""));
+
+        assertEq(weth.balanceOf(creditor), initialCreditorWeth - 1 ether, "Creditor WETH balance after loan acceptance incorrect");
+        assertEq(weth.balanceOf(debtor), initialDebtorWeth + 1 ether, "Debtor WETH balance after loan acceptance incorrect");
+
+        bullaClaim.permitPayClaim({
+            user: debtor,
+            operator: address(bullaFrendLend),
+            approvalType: PayClaimApprovalType.IsApprovedForAll,
+            approvalDeadline: 0,
+            paymentApprovals: new ClaimPaymentApprovalParam[](0),
+            signature: sigHelper.signPayClaimPermit({
+                pk: debtorPK,
+                user: debtor,
+                operator: address(bullaFrendLend),
+                approvalType: PayClaimApprovalType.IsApprovedForAll,
+                approvalDeadline: 0,
+                paymentApprovals: new ClaimPaymentApprovalParam[](0)
+            })
+        });
+
+        // Advance time by 10 days to generate some interest
+        vm.warp(block.timestamp + 10 days);
+        
+        // Make first partial payment
+        uint256 firstPaymentAmount = 0.3 ether;
+        vm.prank(debtor);
```

Comment:
nit: 

you can do
startPrank(debtor);
approve()
pay()
stopPrank();

instead of pranking every line.